deps/happy-eyeballs/happy-eyeballs.c

```diff
@@ -0,0 +1,683 @@
+#include "happy-eyeballs.h"
+#include "util/darray.h"
+#include "util/dstr.h"
+#include "util/platform.h"
+#include "util/threading.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#ifdef _WIN32
+#ifdef _MSC_VER
+#pragma warning(disable : 4996) //depricated warnings
+#endif
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#define GetSockError() WSAGetLastError()
+#ifdef _MSC_VER
+#define snprintf _snprintf
+#endif
+#else // !_WIN32
+#include <netdb.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#define SOCKET int
+#define GetSockError() errno
+#endif
+
+/* happy eyeballs coefficients */
+#define HAPPY_EYEBALLS_DELAY_MS 200 /* this is the same default as libcurl */
+#define HAPPY_EYEBALLS_MAX_ATTEMPTS 6
+#define HAPPY_EYEBALLS_CONNECTION_TIMEOUT_MS \
+	25000 // Total time to wait for sockets, or all finish trying, whichever is shorter
+
+#ifndef INVALID_SOCKET
+#define INVALID_SOCKET ~0
+#endif
+
+#define STATUS_SUCCESS 0
+#define STATUS_FAILURE -1
+#define STATUS_INVALID_ARGUMENT -EINVAL
+
+struct happy_eyeballs_candidate {
+	SOCKET sockfd;
+	os_event_t *socket_completed_event;
+	pthread_t thread;
+	int error;
+};
+
+struct happy_eyeballs_ctx {
+	/**
+	 * socket_fd will be non-zero upon successful connection to the host and port specified.
+	 */
+	int socket_fd;
```

Comment:
```suggestion
	SOCKET socket_fd;
```
Should be the same type as `happy_eyeballs_candidate::sockfd`?